client/app/widgets/Comments/__tests__/Comments.spec.jsx

```diff
@@ -83,41 +82,34 @@ const component = <Comments formProps={formProps} />;
 
 describe('Comments', () => {
   beforeEach(() => {
-    axiosPostSpy = jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({
+    jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({
       data: { comment },
     }));
-    axiosDeleteSpy = jest.spyOn(axios, 'delete').mockImplementation(() => Promise.resolve({
+    jest.spyOn(axios, 'delete').mockImplementation(() => Promise.resolve({
       data: { id },
     }));
   });
 
   it('renders correctly', () => {
-    let wrapper = null;
-    expect(() => {
-      wrapper = mount(component);
-    }).not.toThrow();
-    expect(wrapper).not.toBeNull();
+    render(component);
+    expect(() => component).not.toThrow();
+    expect(component).not.toBeNull();
```

Comment:
Since `component` is defined above, it appears this assertion can't fail, since `component` can never be `null`.

Should this instead be checking against the result of _rendering_ the component, e.g.:

```
const wrapper = render(component);
expect(wrapper).not.toBeNull();
```

But from the docs, I couldn't tell if `wrapper` can ever be `null` either. I tried checking what happens without any arguments to `render` and it still worked.

Would it make more sense to check if the expected child components are rendered instead?

```
it('renders correctly', () => {
    render(component);
    expect(screen.queryByRole('article')).not.toBeInTheDocument(); // no comment yet
    expect(screen.getByRole('button', { name: 'Submit' })).toBeInTheDocument(); // form exists
});
```

Reply:
I am not really familiar with `jest` and `RTL` best practice but I think your suggestion makes more sense based on the test description `it renders correctly`.

Reply:
Awesome! In a rush, I missed that the component also renders a `section` around all comments. That was another 'role' that could be queried. But I think `article` still works 👍